src/dependenpy/algorithms.py

```diff
@@ -0,0 +1,328 @@
+"""dependenpy alogorithms module.
+
+https://dsmweb.org/sequencing-a-dsm/
+"""
+
+from __future__ import annotations
+
+import copy
+from typing import TYPE_CHECKING, Any, List, Tuple
+
+from dependenpy.structures import Matrix
+
+if TYPE_CHECKING:
+    from dependenpy.dsm import DSM, Module, Package
+
+
+class DsmMatrix(Matrix):
```

Comment:
"DsmMatrix" feels a bit redundant since DSM is short for Design Structure Matrix :slightly_smiling_face: